### PR TITLE
CBAPI-1508: Add Asynchronous Query Support and Demonstration on DeviceSearchQuery

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBAPI-1508

## Pull Request Description

Asynchronous query support added to CBC SDK:
* Support for asynchronous executor object in CBCloudAPI
* New AsyncQueryMixin to be added to query objects for asynchronous query support
* Mixin added to DeviceSearchQuery and support added
* list_devices.py example updated to optionally use async query (execute_async() method) via --asynchronous or -Y command line flag

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Unit tests have been added for specific functionality. In addition, list_devices.py has been run both with and without the asynchronous option and the outputs have been verified to be equivalent.

### CODE REVIEW

File: src/cbc_sdk/rest_api.py [hunk 1]
```diff
@@ -250,7 +273,7 @@ def bulk_threat_dismiss(self, threat_ids, remediation=None, comment=None):
         """
         return self._bulk_threat_update_status(threat_ids, "DISMISSED", remediation, comment)
 
-    # ---- ThreathHunter
+    # ---- ThreatHunter
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/rest_api.py:276]: This should be Enterprise EDR

Comment (abowersox-cb) [reply src/cbc_sdk/rest_api.py]: Heh, I fixed the typo, which is the only reason this showed up. :) Perhaps I should add another ticket to do a sweep of the comments for any outdated terminology?

**Review by avanbrunt-cb — APPROVED** [review]
LGTM. Are there any other failure tests cases around async that might happen? or are those hard to trigger?

**Comment by abowersox-cb** [comment]
> Are there any other failure tests cases around async that might happen? or are those hard to trigger?

Well, the implementation as it exists is pretty simple and not particularly error-prone in terms of the code that was written; that said, _anything_ involving multiple threads is going to be hard to trigger failures in.

**Review by llyon-cb — APPROVED** [review]

**Pull request merged**